File: Data/FieldDefinition.cs
using System.Collections.Generic;

namespace MVCCoreVue.Data
{
    public class FieldDefinition
    {
        public object Default { get; set; }
        public bool? Disabled { get; set; }
        public string GroupName { get; set; }
        public string Help { get; set; }
        public bool? HideInTable { get; set; }
        public string Hint { get; set; }
        public string Icon { get; set; }
        public string InputType { get; set; }
        public string Label { get; set; }
        public object Max { get; set; }
        public object Min { get; set; }
        public string Model { get; set; }
        public string Pattern { get; set; }
        public string Placeholder { get; set; }
        public string Prefix { get; set; }
        public bool? Readonly { get; set; }
        public bool? Required { get; set; }
        public int? Rows { get; set; }
        public double? Step { get; set; }
        public string Suffix { get; set; }
        public string Type { get; set; }
        public string Validator { get; set; }
        public List<ChoiceOption> Values { get; set; }
        public bool? Visible { get; set; }
    }

    public class ChoiceOption
    {
        public string Text { get; set; }
        public int Value { get; set; }
    }
}

File: ClientApp/store/field-definition.ts
export interface FieldDefinition {
    autocomplete?: boolean;
    buttons?: Array<any>
    default?: any;
    disabled?: boolean | Function;
    groupName?: string;
    help?: string;
    hideInTable?: boolean | Function;
    hint?: string;
    icon?: string;
    inputType?: string;
    label?: string;
    max?: number;
    min?: number;
    model: string;
    pattern?: string;
    placeholder?: string;
    prefix?: string;
    readonly?: boolean | Function;
    required?: boolean;
    rows?: number;
    step?: number;
    styleClasses?: string | Array<string>;
    suffix?: string;
    type: string;
    validator?: Function | Array<Function>;
    values?: Array<any> | Function;
    visible?: boolean | Function;
}